src/Utils.hs

```diff
@@ -0,0 +1,15 @@
+module Utils (dosedLists, saltedRange) where
+
+
+-- dosedLists 2 [1..6] = [[1,2],[3,4],[5,6]]
+-- dosedLists 3 [1..6] = [[1,2,3],[4,5,6]]
+dosedLists :: (Num a) => Int -> [a] -> [[a]]
```

Comment:
@rexim it's not the same, `chunksOf` doesn't slice last chunk that contains less elements that chunk size. But it doesn't matter when we deal with infinite list, and completely doesn't matter because I refactored `initialState` by @Minoru advise and it doesn't use `dosedLists` or `chunksOf` now.